# Generated by Django 4.2.13 on 2024-07-19 09:33

from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ("moulinette", "0054_alter_moulinetteconfig_criteria_values_and_more"),
    ]

    operations = [
        # So, the initial name of the "activation map" field was "perimeter".
        # When django generates migrations to rename a field, it does not rename
        # the existing indexes, which means there was an index name `something_perimeter_id`
        # When adding a field named "perimeter", we got an error because of an
        # already existing index name.
        migrations.RunSQL(
            """
            ALTER INDEX moulinette_criterion_perimeter_id_5f395795
            RENAME TO moulinette_criterion_activation_map_id_5f395795
        """
        ),
        migrations.AddField(
            model_name="criterion",
            name="perimeter",
            field=models.ForeignKey(
                blank=True,
                null=True,
                on_delete=django.db.models.deletion.PROTECT,
                related_name="criteria",
                to="moulinette.perimeter",
                verbose_name="Perimeter",
            ),
        ),
    ]
